### ISSUE TITLE
Undefined variable $encryption in ImapProtocol.php:73 if encryption is set to false

### ISSUE THREAD

**Issue by ricus-zero2one** [Webklex/php-imap#97]
Hi @Webklex 

webklex/laravel-imap version 2.4.0

(Posting here as code causing the problem is in the underlying php-imap library)

We have some clients who uses `false` as their encryption method for no encryption. We try to encourage everyone to use encryption but in some situations it's not possible.

As far as I can see from the documentation, and how we have been doing it from version 1.x, `false` is an accepted value for encryption when creating a new connection.

However, using the following config:
```php
[
  "username" => "xxxx",
  "password" => "xxxx",
  "host" => "xxxx",
  "port" => 143,
  "encryption" => false,
  "protocol" => "imap",
  "validate_cert" => 0,
]
```
results in an undefined variable `$encryption` error in the ImapProtocol.php class, specifically at line 73.

The function causing this error is the `connect` function, below pulled from vendor:

```php
 public function connect($host, $port = null) {
        $transport = 'tcp';

        if ($this->encryption) { // if $this->encryption is false, then the $encryption variable doesn't get set
            $encryption = strtolower($this->encryption);
            if ($encryption == "ssl") {
                $transport = 'ssl';
                $port = $port === null ? 993 : $port;
            }
        }
        $port = $port === null ? 143 : $port;
        try {
            $this->stream = $this->createStream($transport, $host, $port, $this->connection_timeout);
            if (!$this->assumedNextLine('* OK')) {
                throw new ConnectionFailedException('connection refused');
            }

           //$encryption not defined here if $this->encryption is false, so below throws an exception
            if ($encryption == "tls") { 
                $this->enableTls();
            }
        } catch (\Exception $e) {
            throw new ConnectionFailedException('connection failed', 0, $e);
        }
    }
```

I thought that the problem here could be that I'm assuming the documentation wants boolean `false`, where it should in fact be a string, but I see the constructor also defaults to `false` boolean which would cause the same error would it not?

So to summarise, it seems like if an encryption method of `false` is used, the `$encryption` variable in `ImapProrocol.php` line `73` is not set, so a `ConnectionFailedException` gets thrown.

**Comment by Webklex** [comment]
Hi @ricus-zero2one ,
thanks for your report.

This issue has been resolved by https://github.com/Webklex/php-imap/commit/a2a9c6b452a4f08949f6d80d5629cab4eb13f202 :)

Best regards,

**Comment by ricus-zero2one** [comment]
Hi @Webklex 

That's great to know thanks.

I did run composer update before making this issue. Has a release been tagged with that fix? I see the latest release for php-imap was 10 days ago but that fix was made 7 days ago? Does it maybe still need to be tagged or am I doing something wrong here?

Thanks

**Comment by Webklex** [comment]
Please update to [v2.5.0](https://github.com/Webklex/php-imap/releases/tag/2.5.0). It might take a wile until it is available on packagist etc.

Best regards,

**Comment by ricus-zero2one** [comment]
You are awesome @Webklex ! Thank you so much for the prompt response and tagging the new release so quickly :)
